packages/state-transition/src/cache/epochCache.ts

```diff
@@ -667,10 +667,10 @@ export class EpochCache {
       this.shufflingCache
         .get(epochAfterUpcoming, this.nextDecisionRoot)
         .then((shuffling) => {
-          if (!shuffling) {
-            throw new Error("EpochShuffling not returned from get in afterProcessEpoch");
-          }
-          this.nextShuffling = shuffling;
+          this.nextShuffling = shuffling
+            ? shuffling
+            : // in some spec tests the BeaconChain class is used and it creates a ShufflingCache
+              computeEpochShuffling(state, this.nextActiveIndices, epochAfterUpcoming);
```

Comment:
@twoeths It does not make sense to me why this was required.  The spec test instantiate a `BeaconChain` class which creates a `ShufflingCache`.  It does not have the shufflings though.  I would assume that if the full class is instantiated and the CachedBeaconState that the async build would work normally but it does not.